src/MySqlConnector/Core/ServerSession.cs

```diff
@@ -471,7 +445,11 @@ public async Task DisposeAsync(IOBehavior ioBehavior, CancellationToken cancella
 				ServerVersion = new(initialHandshake.ServerVersion);
 				ConnectionId = initialHandshake.ConnectionId;
 				AuthPluginData = initialHandshake.AuthPluginData;
-				m_useCompression = cs.UseCompression && (initialHandshake.ProtocolCapabilities & ProtocolCapabilities.Compress) != 0;
+				clientCapabilities =
+					HandshakeResponse41Payload.CreateClientCapabilities(initialHandshake.ProtocolCapabilities, cs, m_useCompression,
+						ProtocolCapabilities.Ssl);
```

Comment:
Not sure that it's correct to advertise `ProtocolCapabilities.Ssl` here if a SSL connection may not be attempted.

Reply:
yes, this is a bug